### PR TITLE
Fix storage tests-weekly matrix

### PR DESCRIPTION
**Pull request by weshaggard**
Fix the tests-weekly matrix which has been failing to generate a matrix because the pool names weren't correct.

I also disabled the PrivatePreview configurations because they do not work currently and will require support from the storage team if we want to setup again.

### CODE REVIEW

File: sdk/storage/platform-matrix-all-versions.json [hunk 1]
```diff
@@ -3,11 +3,11 @@
     "Agent": {
       "ubuntu-20.04": {
           "OSVmImage": "env:LINUXVMIMAGE",
-          "Pool": "azsdk-pool-mms-ubuntu-2004-storage"
+          "Pool": "env:LINUXPOOL"
```
Comment (weshaggard) [inline sdk/storage/platform-matrix-all-versions.json:6]: @benbp this is what caused the storage tests-weekly matrix to not generate correctly.

**Review by benbp — APPROVED** [review]

**Pull request merged**